web-registry/src/features/ecocredit/CreateProject/CreateProjectContext.tsx

```diff
@@ -0,0 +1,17 @@
+import { createContext, useContext } from 'react';
+import { DeliverTxResponse } from '@cosmjs/stargate';
+
+type ContextType = {
+  deliverTxResponse?: DeliverTxResponse;
+};
+
+const initialValues = {
+  deliverTxResponse: undefined,
+};
+
+const CreateProjectContext = createContext<ContextType>(initialValues);
+
+const useCreateProjectContext = (): ContextType =>
+  useContext(CreateProjectContext);
```

Comment:
Might want to wrap these in a helper function to throw an error if they're not used in the right context. Something like
```suggestion
const useCreateProjectContext = (): ContextType => {
  const context = useContext(CreateProjectContext);
  if (context === undefined) {
    throw new Error(
      'useCreateProjectContext must be used within a ProjectProvider',
    );
  }
};
```

Reply:
I'm not sure that is necessary. useCreateProjectContext is a hook - I don't think it needs a provider. But maybe I'm missing something.

Reply:
hmm, looking at the code a bit more, I'm not sure we're using context here appropriately -  you can only share state with react context if it's nested underneath the provider for that context (like providers in `index.tsx`)

it was easier to show this with code than type it out, so made a small PR into yours: https://github.com/regen-network/regen-web/pull/1078